File: qycms-blog/blog-provider/src/main/java/com/windcoder/qycms/blog/entity/Comment.java
package com.windcoder.qycms.blog.entity;

import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Lob;

public class Comment {
    @Id
    @GeneratedValue
    private Long id;
    @Lob
    private String content;

    /**
     * 评论者邮件
     */
    private String mail;

    /**
     * 评论者网址
     */
    private String url;

    /**
     * 评论者ip地址
     */
    private String ip;

    /**
     * 评论状态
     */
    private String status;

    /**
     * 评论者客户端
     */
    private String agent;

    private Comment parent;
}

File: src/app/comment/entity/comment.ts
import { CommentAgent } from "./comment-agent";
export type CommentStatus = 'APPLIED' | 'ENROLLED' | 'REFUSED' ;


export class Comment {
    id: number;
    content: string;
    author: string;
    email: string;
    url: string;
    ip: string;
    status: CommentStatus;
    target: CommentAgent;
    depth: number;
    parent: Comment;
    topParent: Comment;
    replyCount: number;
    createdDate: Date;
}
